src/core/impl/comp_node/cuda/comp_node.cpp

```diff
@@ -637,7 +637,7 @@ CompNode::Impl* CudaCompNode::load_cuda(
     if (!available_node) {
         mgb_assert(sd.nr_node < sd.MAX_NR_COMP_NODE,
                 "too many CompNode allocated");
-        mgb_assert(locator.device < sd.MAX_NR_COMP_NODE,
+        mgb_assert(locator.device < sd.MAX_NR_DEVICE,
```

Comment:
Actually, this line should be placed outside to assert `locator.device < sd.MAX_NR_DEVICE` when `bool(avaiable_node) = true`.

Maybe put it to L644 would be better.

Reply:
Hi @xxr3376 I have modified the code as you say,
but it failed when building CI(GPU test)
https://github.com/MegEngine/MegEngine/pull/47

Reply:
See https://github.com/MegEngine/MegEngine/pull/47#issuecomment-626851313